gef-ui/api-tests/bank-review-date.post.api-test.js

```diff
@@ -0,0 +1,194 @@
+
+api.getFacility = jest.fn();
+api.getApplication = jest.fn();
+api.updateFacility = jest.fn();
+api.updateApplication = jest.fn();
+
+const dealId = '123';
+const facilityId = '111';
+
+describe('bank-review-date routes', () => {
+  beforeEach(async () => {
+    resetAllWhenMocks();
+    await storage.flush();
+  });
+
+  afterAll(async () => {
+    resetAllWhenMocks();
+    jest.resetAllMocks();
+    await storage.flush();
+  });
+
+  describe('POST /application-details/:dealId/facilities/:facilityId/bank-review-date', () => {
+    beforeEach(() => {
+      when(api.getFacility)
+        .calledWith({ facilityId, userToken: expect.anything() })
+        .mockResolvedValueOnce({
+          details: {
+            isUsingFacilityEndDate: null,
+            coverStartDate: '2024-07-15T00:00:00.000Z',
+          },
+        });
+    });
+
+    describe('with saveAndReturn false', () => {
+      withRoleValidationApiTests({
+        makeRequestWithHeaders: (headers) =>
+          postBankReviewDateWithHeaders({ body: { 'bank-review-date-year': '2024', 'bank-review-date-month': '08', 'bank-review-date-day': '12' }, headers }),
+        whitelistedRoles: [MAKER],
+        successCode: 302,
+        successHeaders: {
+          location: `/gef/application-details/${dealId}/facilities/${facilityId}/provided-facility`,
+        },
+      });
+    });
+
+    describe('with saveAndReturn true', () => {
+      withRoleValidationApiTests({
+        makeRequestWithHeaders: (headers) =>
+          postBankReviewDateWithHeaders({
+            body: { 'bank-review-date-year': '2024', 'bank-review-date-month': '08', 'bank-review-date-day': '12' },
+            headers,
+            saveAndReturn: true,
+          }),
+        whitelistedRoles: [MAKER],
+        successCode: 302,
+        successHeaders: {
+          location: `/gef/application-details/${dealId}`,
+        },
+      });
+    });
+
+    describe('when the user is a maker', () => {
+      let sessionCookie;
+
+      beforeEach(async () => {
+        ({ sessionCookie } = await storage.saveUserSession([MAKER]));
+      });
+
+      it('returns 200 if the request body fails validation', async () => {
+        // Arrange
+        const body = { 'bank-review-date-year': '2023', 'bank-review-date-month': '8', 'bank-review-date-day': '12' };
+
+        // Act
+        const response = await postBankReviewDateWithBodyAndSessionCookie({ body, sessionCookie });
+
+        // Assert
+        expect(response.status).toBe(200);
+      });
+
+      it('returns 200 if the request body fails validation & saveAndReturn = true', async () => {
+        // Arrange
+        const body = { 'bank-review-date-year': '2023', 'bank-review-date-month': '8', 'bank-review-date-day': '12' };
+
+        // Act
+        const response = await postBankReviewDateWithBodyAndSessionCookie({ body, sessionCookie, saveAndReturn: true });
+
+        // Assert
+        expect(response.status).toBe(200);
```

Comment:
do we want to check api is not called? this feels slightly pointless as a test atm